OpenRVS/classes/OpenMultiPlayerWidget.uc

```diff
@@ -308,22 +329,219 @@ function Refresh(bool bActivatedByUser)
+				CurServer.bSameVersion = false;
+			else
+				CurServer.bSameVersion = true;
+			CurServer.bLocked = bSvrLocked;//1.5 added locked here
+			//1.5 add ping
+			i = Timer.EndTimer(CurServer.szIPAddr);
+			if ( i != -1 )
+				CurServer.iPing = i / 2;//divide by two for single-trip time - consistent with how Ubi used to measure
+			CurServer = none;//break the while loop
 		}
-		Timer.StartTimer(CurServer.szIPAddr);
-		OpenClientBeaconReceiver(m_GameService.m_ClientBeacon).QuerySingleServer(self,
-			Left(CurServer.szIPAddr,InStr(CurServer.szIPAddr,":")),
-			int(Mid(CurServer.szIPAddr,InStr(CurServer.szIPAddr,":")+1))+1000);
+		else
 		CurServer = R6WindowListServerItem(CurServer.Next);
+	}*/
+	
+	//1.5
+	//modify the servers array rather than the list items
+	//then rebuild list items with GetGSServers()
+	i = 0;
+	while ( i < OpenQueries.length )
+	{
+		if ( OpenQueries[i] == sIP )//close open query
+		{
+			OpenQueries.remove(i,1);
+			break;
+		}
+		i++;
 	}
+	i = 0;
+	while ( i < ServerList.length )
+	{
+		if ( ServerList[i].IP == sIP )
+		{
+			ServerList[i].ServerName = sSvrName;
+			ServerList[i].Locked = bSvrLocked;
+			ServerList[i].iMaxPlayers = iMaxP;
+			ServerList[i].iNumPlayers = iNumP;
+			ServerList[i].szGameType = GetLevel().GetGameNameLocalization(sGMode);
+			ServerList[i].szMap = sMapName;
+			ServerList[i].bWrongVersion = ( caps(class'Actor'.static.GetModMgr().m_pCurrentMod.m_szKeyWord) != caps(sModName) );
+			iTime = Timer.EndTimer(sIP);
+			if ( iTime != -1 )
+				ServerList[i].iPing = iTime / 2;//divide by two for single-trip time - consistent with how Ubi used to measure
+		}
+		i++;
+	}
+	GetGSServers();
+}
+
+// InitServerList() creates a window for the server list. It is called by
+// ShowWindow(). In our version, we are able to override the server ping timeout.
+// Overrides the matching function in R6MenuMultiPlayerWidget.
+// 1.3: fixed access none
+function InitServerList()
+{
+	local Font buttonFont;
+	local int iFiles, i, j;
+
+	// Create window for server list
+	if ( m_ServerListBox != none )
+		return;
+	m_ServerListBox = R6WindowServerListBox(CreateWindow(class'R6WindowServerListBox',K_XSTARTPOS_NOBORDER,K_YPOS_FIRST_TABWINDOW,K_WINDOWWIDTH_NOBORDER,K_FFIRST_WINDOWHEIGHT,self));
+	m_ServerListBox.Register( m_pFirstTabManager);
+	m_ServerListBox.SetCornerType(No_Borders);
+	m_ServerListBox.m_Font = Root.Fonts[F_ListItemSmall];
+	if ( m_LanServers != none )//accessed none fix?
+		m_ServerListBox.m_iPingTimeOut = m_LanServers.NativeGetPingTimeOut();
+	else
+		m_ServerListBox.m_iPingTimeOut = 10000;
+}
+
+//sorting update 1.5
+//calls super if in LAN tab - use the built-in native functions for LAN
+function ResortServerList(int iCategory, bool _bAscending)
+{
+	local int i,j;
+	local bool bSwap;
+	local int iListSize;
+	local AServer temp;
+	local string sCompare1,sCompare2;
+	local int iCompare1,iCompare2;
+	local bool bIntComp;
+
+	if ( m_ConnectionTab != TAB_Lan_Server )

```

Comment:
Would flip this around to avoid nesting the entire block inside the if statement:
```
if tab_lan_server:
  super.foo()
  return

other code here
```
```